pwa/src/templates/templateParts/download/DownloadTemplate.tsx

```diff
@@ -1,63 +1,80 @@
 import { faArrowLeft, faDownload } from "@fortawesome/free-solid-svg-icons";
 import { useTranslation } from "react-i18next";
-import clsx from "clsx";
 import { useGatsbyContext } from "../../../context/gatsby";
 
 interface DownloadTemplateProps {
-  icon: JSX.Element;
-  label: string;
-  sizeKb: string;
+  items: DownloadProps[];
+  backUrl: string;
+  icon?: JSX.Element;
   layoutClassName?: string;
 }
 
-export const DownloadTemplate: React.FC<DownloadTemplateProps> = ({ layoutClassName, icon, label, sizeKb }) => {
+export interface DownloadProps {
+  label: string;
+  type: string;
+  size: string;
+  downloadLink: string;
+}
+
+export const DownloadTemplate: React.FC<DownloadTemplateProps> = ({ items, backUrl, icon, layoutClassName }) => {
   const { t } = useTranslation();
   const { screenSize } = useGatsbyContext();
+  const [isVisible, setIsVisible] = React.useState<any>({});
 
-  const NotificationPopUpController = _NotificationPopUp.controller;
-  const NotificationPopUp = _NotificationPopUp.NotificationPopUp;
+  const toggleVisibility = (label: string) => {
+    setIsVisible({
+      [label]: {
+        value: true,
+      },
+    });
+  };
 
-  const { isVisible, show, hide } = NotificationPopUpController();
+  const NotificationPopUp = _NotificationPopUp.NotificationPopUp;
 
   return (
     <div className={clsx([layoutClassName && layoutClassName], screenSize === "mobile" && styles.downloadName)}>
-      <DownloadCard
-        label={label}
-        sizeKb={sizeKb}
-        downloadLabel="Download"
-        icon={icon}
-        handleClick={() => {
-          show();
-        }}
-      />
-
-      {isVisible && (
-        <div className={styles.overlay}>
-          <NotificationPopUp
-            {...{ hide, isVisible }}
-            title={`${t("Warning")}!`}
-            description={t(
-              "This file comes from a 3rd party and can potentially be harmfull for your PC. Are you sure you want to download this?",
-            )}
-            primaryButton={{
-              label: "Download",
-              icon: <FontAwesomeIcon icon={faDownload} />,
-              handleClick: () => ({}),
-            }}
-            secondaryButton={{
-              label: t("Go back"),
-              icon: <FontAwesomeIcon icon={faArrowLeft} />,
-              handleClick: () => ({}),
+      {items.map(({ label, size, type, downloadLink }, idx) => (
+        <React.Fragment key={idx}>
+          <DownloadCard
+            label={label ?? downloadLink}
+            type={type ?? downloadLink.substring(downloadLink.lastIndexOf(".") + 1)}
+            {...{ size, icon }}
+            handleClick={() => {
+              toggleVisibility(label ?? downloadLink);
             }}
-            layoutClassName={styles.popup}
           />
-        </div>
-      )}
+
+          {isVisible[label ?? downloadLink]?.value && (
+            <div className={styles.overlay}>
+              <NotificationPopUp
```

Comment:
I think we should refactor the placement, and animation, of this pop-up. Check out the Rating pop-up, this has already been updated.

Note: I now see that the Rating pop-up does not always center correctly, maybe you can fix that in this pull request (if it's easy to fix).